### DISCUSSION TITLE
Question about persistent configuration

### DISCUSSION THREAD

**Discussion by Steamgjk** [Derecho-Project/derecho#234] [Q&A] [answered]
I am now trying to use ramfs as the storage backend to store version-based data.
There are two attributes in the cfg file.

file_path = .plog
ramdisk_path = /dev/shm/volatile_t

What is the relation between them. 
I have created the ramfs as the path  /mnt/ram/

So should set ramdisk_path as  /mnt/ram  or set file_path as  /mnt/ram

I guess  file_path just specifies the filename, so I should set  ramdisk_path=/mnt/ram, and leave file_path as .plog

Then If I run derecho, the versioned data should be stored in /mnt/ram/.plog, is that right?

**Comment by songweijia** [comment]
`ramdisk_path` is for the location of storage type ST_MEM. By default, if you don't specify the storage type, Persistent<T> will use ST_FILE, which will store the data in the file system (controlled by file_path).

**Reply by Steamgjk** [reply]
Hi, @songweijia . I intend to use ramfs as suggested by https://github.com/Derecho-Project/derecho/discussions/233

In that way, do I need to touch these two fields, or not?

Besides, could you help to take a look at my crash issue https://github.com/Derecho-Project/derecho/issues/235

Derecho is awesome, but it may still take some effort for a freshman like me to get used to it.

**Comment by songweijia** [comment, marked as answer]
In #233, you are not going to use ST_MEM. Therefore all you need to do to use ramfs is to point `file_path` to `/mnt/ram`.
